### ISSUE TITLE
Placing assets in subdirectories breaks LiveReload

### ISSUE THREAD

**Issue by dcalhoun** [robwierzbowski/generator-jekyllrb#56]
### TL;DR

Jekyll's exclude option in `_config.yml` doesn't appear to accept subdirectories.
### Details

The default configuration for the JavaScript directory is `js`. If during the configuration [you choose to place it in a subdirectory](https://github.com/robwierzbowski/generator-jekyllrb/blob/master/app/index.js?source=c#L160), say `assets/js`, it breaks the LiveReload of the JavaScript.

From my testing, it seems that Jekyll's `_config.yml` exclude option does not accept paths to subdirectories. You may only pass file or folder names. Jekyll is ignoring requests to exclude `assets/js` and the JavaScript (and any other asset in a subdirectory) is copied to the `.jekyll` directory. The [server looks](https://github.com/robwierzbowski/generator-jekyllrb/blob/master/app/templates/Gruntfile.js?source=c#L74-L75) in the `.jekyll` directory for assets prior to looking in `app`.

This isn't an issue when using CoffeeScript compilation as the CoffeeScript files are compiled into `.tmp`. The server looks in `.tmp` before looking into `.jekyll` or `app`, so it ignores the fact that the JavaScript files are in `.jekyll`. **Note:** I haven't tested it, but I'm would imagine this would happen with CSS v Sass as well.
### Possible Solutions
1. Ask the user if they would like to place all the assets in a subdirectory. If they provide a directory to place all the assets, then add that directory to the exclude array rather than each asset directory.
2. Copy the JavaScript from `app` to `.tmp` on save, as the server would look there first.
3. Not allow subdirectories for assets. :frowning: 

Thoughts?
